### ISSUE TITLE
Repo init

### ISSUE THREAD

**Issue by ZenGround0** [FILCAT/pdp#1]
# Setup repo
Top level should contain build / deploy script, utils folder, solidity project folder.

## Solidity
We're going to use foundry since it is the least worst of our options. 
 We need to decide on solidity compiler version.  We should set up remappings file, potentially bring in some basic dependencies that we expect to use just to get the dependency flow working right.

## Building
Deploy script can depend on lotus since we are targeting Filecoin L1 to start.  For now we can add `make deploy` to makefile

Setup testing directory and `make test`

**Comment by ZenGround0** [comment]
It still remains to get CI working in the repo

**Comment by ZenGround0** [comment]
Will be looking into ipdx unified CI here: https://github.com/ipdxco/unified-github-workflows?tab=readme-ov-file#usage

**Comment by ZenGround0** [comment]
Used github action CI.  Pushed directly to main for debugging purposes.  Commit getting things woking here: https://github.com/FILCAT/pdp/commit/8b6978d9dc39efaa685e8805d88ee21071ecabf1

Unfortunately it's a bit more brittle than I would like since I needed to make the configure script aware of github actions pathing because foundry's writing to and sourcing of the bash rc was glitching.  But CI now working and it will be obvious if it breaks.
